src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/ViewComponentTagHelperTargetExtension.cs

```diff
@@ -162,7 +164,7 @@ private string[] GetMethodParameters(TagHelperDescriptor tagHelper)
         {
             var propertyNames = tagHelper.BoundAttributes.Select(attribute => attribute.GetPropertyName());
             var joinedPropertyNames = string.Join(", ", propertyNames);
-            var parametersString = $"new {{ { joinedPropertyNames } }}";
+            var parametersString = $"{TagHelperContextVariableName}.{TagHelperContextAttributesVariableName}.ToDictionary(attr => attr.Name, attr => attr.Value)";
```

Comment:
Super smart way of doing this. Basically relies on the matching conventions / TagHelper internals to do all the heavy lifting 👏, love it!

One thing to note is that the TagHelperContext will contain every attribute associated with the element. Meaning if you have

```C#
public IViewComponentResult Invoke(string abc = "hello")
```

and you have:
```
<vc:test Abc="Value1" abc="Value2" ABc="Value3" class="bar" />
```

TagHelperContext.AllAttributes will be equivalent to:
```C#
new Dictionary<string, object>
{
    ["Abc"] = "Value1",
    ["abc"] = "Value2",
    ["ABc"] = "Value3",
    ["class"] = "bar",
}
```

Are the extra values "ok" to the runtime impl?

Reply:
Yep -- these parameters are OK. They get interpreted as HTMLAttributes instead of parameters.